src/cmd/key/renew.sh

```diff
@@ -28,11 +28,14 @@ Try first:  $(basename $0) key join
     local today=$(date -d $(date +%F) +%s)
     time=$(( ( $expday - $today ) / 86400 ))
 
-    local commands=";expire;$time;y;key 1;expire;$time;y;key 1;save"
-    commands=$(echo "$commands" | tr ';' "\n")
-    echo -e "$commands" | gpg --no-tty --command-fd=0 --key-edit $GPG_KEY 2>/dev/null
+    # blank quotes ("") => None
+    local commands=("" "expire" "" "$time" "" "key 1" "" "expire" "" "$time" "" "save" "" "")
+    call_gpg fn/interact.py "$GPG_KEY" "${commands[@]}"
     call_fn gpg_send_keys $GPG_KEY
 
+    err=$?
+    [[ $err == 0 ]] || fail "error renewing $GPG_KEY"
+
```

Comment:
Why don't you print this message from the Python code, if there is some error?

Reply:
In most of the cases there is no harm. In many cases the bash script may need to clean up even if the python call returns an error status. So, automatic termination of execution is not a good option.
Let the bash script check for the exit status explicitly, if needed, and do the right action. In most of the cases nothing needs to be done.